### PR TITLE
1-Setup-Data-Layer: Initial commit of required models and Api calls. …

### PR DESCRIPTION
**Pull request by DurtDiv3r**
Initial commit of required models and Api calls. Additional: Added ktlint to build.gradle.

Clean Architecture:
https://medium.com/android-dev-hacks/detailed-guide-on-android-clean-architecture-9eab262a9011

Wrapper class:
https://medium.com/codex/kotlin-sealed-classes-for-better-handling-of-api-response-6aa1fbd23c76

### CODE REVIEW

File: app/src/main/java/com/islaharper/dawnofandroid/domain/model/ApiRequest.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.islaharper.dawnofandroid.domain.model
+
+import kotlinx.serialization.Serializable
+
+// Used to verify token for Google One Tap signin
+@Serializable
+data class ApiRequest(
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/domain/model/ApiRequest.kt]: This class is named `ApiRequest` , which makes it sound like it is the base class for all other ApiRequests, but it currently only seems to be used for making a token verification request. What do you think about finding a clearer name for this one?

File: app/src/main/java/com/islaharper/dawnofandroid/domain/model/ApiResponse.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.islaharper.dawnofandroid.domain.model
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
+
+@Serializable
+data class ApiResponse(
```
Comment (drawers) [inline app/src/main/java/com/islaharper/dawnofandroid/domain/model/ApiResponse.kt:6]: We discussed a solution in the other project using sealed classes. This current `ApiResponse` class allows for `success = true` and `error` to be non-null. We want to define invalid states like `success = true` and a non-null error out of existence. Can you please look at the previous discussion and have another go at modeling the `ApiResponse` here?

File: app/build.gradle [hunk 3]
```diff
@@ -54,6 +55,16 @@ android {
     }
 }
 
+ktlint {
```
Comment (drawers) [inline app/build.gradle:58]: Thank you for adding ktlint.

**Review by drawers — COMMENTED** [review]
A good start. As discussed in the online meeting today, could we please add the course or article we are basing this implementation on so that I have more context on how to review.

File: app/src/main/java/com/islaharper/dawnofandroid/util/Resource.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.islaharper.dawnofandroid.util
+
+// Handles ApiResponse
+sealed class Resource<out Any> {
+    object Idle : Resource<Nothing>()
```
Comment (DurtDiv3r) [inline app/src/main/java/com/islaharper/dawnofandroid/util/Resource.kt:5]: Example of when Idle is used:
`    private val _apiResponse: MutableState<Resource<ApiResponse>> = mutableStateOf(Resource.Idle)
    val apiResponse: State<Resource<ApiResponse>> = _apiResponse`

It's an addition to the Resource wrapper and is used as an initial state.

https://www.udemy.com/course/secure-user-authentication-android-ktor-server-mongodb/learn/lecture/31216110#overview

**Review by drawers — APPROVED** [review]

**Pull request merged**